rllib/evaluation/sample_batch_builder.py

```diff
@@ -151,6 +152,11 @@ def postprocess_batch_so_far(self, episode):
                     "from a single trajectory.", pre_batch)
             post_batches[agent_id] = policy.postprocess_trajectory(
                 pre_batch, other_batches, episode)
+            # Call the Policy's Exploration's postprocess method.
+            if hasattr(policy, "exploration") and isinstance(
+                    policy.exploration, Exploration):
```

Comment:
Don't we require all policies have exploration defined?

Reply:
(We seem to assume this in sampler.py below)

Reply:
Do we? Originally, we said yes, but then we discussed briefly last week that this may not be the case.
In our doc, it says that every policy has one.
Ok, let's assume that this is the case. Every Policy has an Exploration.

Reply:
Done.

Reply:
Yeah not sure... it seems reasonable if everyone calls super().__init__() on policy.

Reply:
Totally. The problem is that everyone is doing that at a different position in their (custom) c'tors, sometimes at the very end (not-preferred) sometimes at the beginning (preferred).
That's why it's so important to do the super() call as early as possible. Then the rest of the constructor can assume that the super props are there already. This is not the case e.g. for DynamicTFPolicy and e.g. DDPG.  We should fix this in the "Policy cleanup" PR ahead.